### DISCUSSION TITLE
Enabling the https protocol

### DISCUSSION THREAD

**Discussion by mike-lischke** [preactjs/wmr#946] [Q&A] [answered]
I started evaluating WMR to port a large React application over to it and created a project folder from scratch following [this step list](https://wmr.dev/docs#manual-installation-and-setup). Everything went fine and now I want to use the https protocol, but cannot find a way to set this up. Apparently there's no setting for this protocol and I couldn't find any documentation or help (searched the WMR site, this repository, Stackoverflow.com and finally the web). Here's my WMR config:

```typescript
import { defineConfig } from 'wmr';

// Full list of options: https://wmr.dev/docs/configuration
export default defineConfig((options) => {
    return {
        alias: {
            react: 'preact/compat',
            'react-dom': 'preact/compat'
        },
        out: "build",
        host: "localhost",
        port: 3301,
        sourcemap: true,
        debug: true,
    };
});
```

I tried options like `https: true` and `protocol: "https"`, but that didn't help.

Anyone here to help a complete newbie with WMR to get this task done?

**Comment by rschristian** [comment, marked as answer]
It's less than ideal naming, but the `--http2` flag is in all reality a https flag. `createSecureServer` returns a `tls.Server` instance and, through its options, supports downgrading clients that are not able to support http2:

https://github.com/preactjs/wmr/blob/ab4b512875ae63273cfbd9c4c5972aadd3218bac/packages/wmr/src/lib/http2.js#L4-L16

**Reply by mike-lischke** [reply]
Interesting, didn't came to my mind that http2 would mean https here :-) Thanks for the answer!

**Reply by rschristian** [reply]
HTTP/2 does necessitate HTTPS for what it's worth, seeing as how no browser supports HTTP/2 without it. But yep, poor naming in the case of someone wanting HTTPS and not necessarily caring about HTTP/2. Sorry about that!
